extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +148,223 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!Window} win The window element
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {function(!string,
+   *   !../../../src/3p-analytics-common.AmpAnalytics3pResponse)=}
+   *   opt_processResponse An optional function to receive any response
+   *   messages back from the cross-domain iframe
+   */
+  processCrossDomainIframe(win, transportOptions, opt_processResponse) {
+    const frameUrl = dev().assertString(transportOptions['iframe'],
+        'Cross-domain frame parameters missing ${this.type_}');
+    let frameData;
+    let shouldAddToDOM = false;
+    if (Transport.hasCrossDomainIframe(frameUrl)) {
+      frameData = Transport.getFrameData(frameUrl);
+      Transport.incrementIframeUsageCount_(
+        /** @type{!FrameData} */ (frameData));
+    } else {
+      frameData = this.createCrossDomainIframe(win, frameUrl,
+          opt_processResponse);
+      shouldAddToDOM = true;
+    }
+    dev().assert(frameData, 'Trying to use non-existent frame');
+    if (transportOptions['extraData']) {
+      frameData.queue.setExtraData(this.id_, transportOptions['extraData']);
+    }
+    if (shouldAddToDOM) {
+      win.document.body.appendChild(frameData.frame);
+    }
+  }
+
+  /**
+   * Create a cross-domain iframe for third-party vendor anaytlics
+   * @param {!Window} win The window element
+   * @param {!string} frameUrl  The URL of the cross-domain iframe
+   * @param {function(!string,
+   *   !../../../src/3p-analytics-common.AmpAnalytics3pResponse)=}
+   *   opt_processResponse An optional function to receive any response
+   *   messages back from the cross-domain iframe
+   * @return {!FrameData}
+   * @VisibleForTesting
+   */
+  createCrossDomainIframe(win, frameUrl, opt_processResponse) {
+    const sentinel = Transport.createUniqueId_();
+    const scriptSrc = getMode().localDev
+      ? 'dist.3p/current/ampanalytics-lib.js'
+      : '$internalRuntimeVersion$/ampanalytics-v0.js';
+    const frameName = JSON.stringify(/** @type {JsonObject} */ ({
+      scriptSrc,
+      sentinel,
+    }));
+    const frame = createElementWithAttributes(win.document, 'iframe',
+        /** @type {!JsonObject} */ ({
+          sandbox: 'allow-scripts allow-same-origin',
+          name: frameName,
+          src: frameUrl,
+          'data-amp-3p-sentinel': sentinel,
```

Comment:
I believe this is indeed necessary, because of https://github.com/ampproject/amphtml/blob/master/src/iframe-helper.js#L378